core/consensus/babe/src/lib.rs

```diff
@@ -464,12 +486,13 @@ fn check_header<B: Block + Sized, C: AuxStore>(
 		babe_err!("Header {:?} has a bad seal", hash)
 	})?;
 
-	let pre_digest = find_pre_digest::<B>(&header)?;
-	let BabePreDigest { slot_num, index, ref proof, ref vrf_output } = pre_digest;
+	let (_epoch, pre_digest) = find_pre_digest::<B>(&header)?;
```

Comment:
Shouldn't we be checking wether this is the expected epoch?

Reply:
We should.  Good catch!

Reply:
This will be caught by the runtime (which executes on each full node).